src/domTree.js

```diff
@@ -211,6 +205,33 @@ class span<ChildType: VirtualNodeInterface> extends HtmlDomContainer<ChildType>
     }
 }
 
+// NOTE: Defining both DomSpan and SvgSpan type directly as span<HtmlDomNode> and
+// span<svgNode> seems to confuse flow when dealing with the type DomSpan|SvgSpan.
+
+/** Span wrapping other DOM nodes. See NOTE above. */
+export class DomSpan extends span<HtmlDomNode> {
```

Comment:
The polyfill of class inheritance is expensive. #1469 is pending review, but I think it's best to avoid class inheritance.

Reply:
Ok, after the documentFragment refactor per below, I was able to undo this without flow being confused again.